drivers/mrf24j40/mrf24j40_getset.c

```diff
@@ -0,0 +1,499 @@
+        case 11:    channel_value = 0x03;
+            break;
+        case 12:    channel_value = 0x13;
+            break;
+        case 13:    channel_value = 0x23;
+            break;
+        case 14:    channel_value = 0x33;
+            break;
+        case 15:    channel_value = 0x43;
+            break;
+        case 16:    channel_value = 0x53;
+            break;
+        case 17:    channel_value = 0x63;
+            break;
+        case 18:    channel_value = 0x73;
+            break;
+        case 19:    channel_value = 0x83;
+            break;
+        case 20:    channel_value = 0x93;
+            break;
+        case 21:    channel_value = 0xa3;
+            break;
+        case 22:    channel_value = 0xb3;
+            break;
+        case 23:    channel_value = 0xc3;
+            break;
+        case 24:    channel_value = 0xd3;
+            break;
+        case 25:    channel_value = 0xe3;
+            break;
+        case 26:    channel_value = 0xf3;
+            break;
+        default:
+            channel_value = 0x3;
+    }
+
+    mrf24j40_reg_write_long(dev, MRF24J40_REG_RFCON0, channel_value);
+    /*
+     * Note:Perform an RF State Machine Reset (see Section 3.1 �Reset�)
+     * after a channel frequency change. Then, delay at least 192 us after
+     * the RF State Machine Reset, to allow the RF circuitry to calibrate.
+     */
+    mrf24j40_reset_state_machine(dev);
+}
+
+uint16_t mrf24j40_get_pan(mrf24j40_t *dev)
+{
+    return dev->netdev.pan;
+}
+
+void mrf24j40_set_pan(mrf24j40_t *dev, uint16_t pan)
+{
+    le_uint16_t le_pan = byteorder_btols(byteorder_htons(pan));
+
+    dev->netdev.pan = pan;
+    DEBUG("pan0: %u, pan1: %u\n", le_pan.u8[0], le_pan.u8[1]);
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_PANIDL, le_pan.u8[0]);
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_PANIDH, le_pan.u8[1]);
+}
+
+
+int16_t mrf24j40_get_txpower(mrf24j40_t *dev)
+{
+    uint8_t txpower;
+
+    txpower = (mrf24j40_reg_read_long(dev, MRF24J40_REG_RFCON3) >> 3) & 0x1F;
+    return tx_pow_to_dbm[txpower];
+}
+
+void mrf24j40_set_txpower(mrf24j40_t *dev, int16_t txpower)
+{
+    uint8_t txpower_reg_value;
+
+    /* positive values are better with a conversion array */
+    txpower = (txpower < 0) ? -1 * txpower : txpower;
+    txpower = (txpower > 36) ? 36 : txpower;
+    txpower_reg_value = dbm_to_tx_pow[txpower];
+
+    mrf24j40_reg_write_long(dev, MRF24J40_REG_RFCON3, txpower_reg_value);
+
+}
+
+uint8_t mrf24j40_get_max_retries(mrf24j40_t *dev)
+{
+    return MRF24J40_MAX_FRAME_RETRIES;
+}
+
+uint8_t mrf24j40_get_csma_max_retries(mrf24j40_t *dev)
+{
+    uint8_t tmp;
+
+    tmp  = mrf24j40_reg_read_short(dev, MRF24J40_REG_TXMCR);
+    tmp  &= 0x7;
+    return tmp;
+}
+
+void mrf24j40_set_csma_max_retries(mrf24j40_t *dev, int8_t retries)
+{
+    uint8_t tmp;
+
+    /* get current register settings */
+    tmp  = mrf24j40_reg_read_short(dev, MRF24J40_REG_TXMCR);
+    /* clear csma bits */
+    tmp  &= ~(0x7);
```

Comment:
Yes, Line 304 does an `OR` operation with the new settings so that the other settings in that register are not affected. The clear is needed so that an old CSMA setting does not affect the new setting. For example, `0b100` is for 4 backoffs. Changing to 3 (`0b011`) backoffs without this clear would result in accidentaly setting `111` for the CSMA bits in the register.